Libraries/Blob/RCTBlobManager.m

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 2015-2016 Silk Labs, Inc.
+ * All Rights Reserved.
+ */
+
+#import "RCTBlobManager.h"
+
+#import <React/RCTConvert.h>
+
+static NSString *const kBlobUriScheme = @"blob";
+
+@implementation RCTBlobManager
+{
+  NSMutableDictionary<NSString *, NSData *> *_blobs;
+  NSOperationQueue *_queue;
+}
+
+RCT_EXPORT_MODULE(BlobModule)
+
+- (NSDictionary<NSString *, id> *)constantsToExport
+{
+  return @{
+    @"BLOB_URI_SCHEME": kBlobUriScheme,
+    @"BLOB_URI_HOST": [NSNull null],
+  };
+}
+
+- (NSString *)store:(NSData *)data
+{
+  NSString *blobId = [NSUUID UUID].UUIDString;
+  [self store:data withId:blobId];
+  return blobId;
+}
+
+- (void)store:(NSData *)data withId:(NSString *)blobId
+{
+  if (!_blobs) {
+    _blobs = [NSMutableDictionary new];
+  }
+  _blobs[blobId] = data;
```

Comment:
Since this is just a port of existing work, haven't looked into that before that comment. Thanks for bringing it up. Certainly these methods are not only executed by the JS calls, but also from within the Websocket module. 

I think wrapping entire method in a `synchronized(_blobs)` would do the trick?